File: objectiveai-js/src/functions/inventions/response/streaming/object.ts
import { z } from "zod";

export const FunctionsInventionsResponseStreamingObjectSchema = z.enum(["alpha.scalar.function.invention.chunk","alpha.vector.function.invention.chunk"]).meta({ title: "functions.inventions.response.streaming.Object" });
export type FunctionsInventionsResponseStreamingObject = z.infer<typeof FunctionsInventionsResponseStreamingObjectSchema>;

File: objectiveai-rs/src/functions/inventions/response/streaming/object.rs
use serde::{Deserialize, Serialize};
use schemars::JsonSchema;

#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize, JsonSchema)]
#[schemars(rename = "functions.inventions.response.streaming.Object")]
pub enum Object {
    #[serde(rename = "alpha.scalar.function.invention.chunk")]
    AlphaScalarFunctionInventionChunk,
    #[serde(rename = "alpha.vector.function.invention.chunk")]
    AlphaVectorFunctionInventionChunk,
}
